docs/source/topics/debug.rst

```diff
@@ -0,0 +1,105 @@
+Usage Assertions and Performance
+================================
+
+In order to help writing correct simulations,
+μSim uses a wide range of usage assertions.
+These generally verify argument values and types in advance,
+so that you can pinpoint where errors originate.
+
+At the same time, μSim strives to provide
+best performance for correct simulations.
+Once you have verified your simulation,
+you can elide all consistency assertions with zero cost.
+
+:note: μSim runs in assertion mode by default.
+
+Assertions and Debug Extensions
+-------------------------------
+
+μSim uses assertions when checking
+the internal logic of simulations.
+For example, :term:`time` in a simulation must always advance
+into the future -
+assertions protect against creating events in the past:
+
+.. code:: python3
+
+    >>> from usim import time, run
+    >>>
+    >>> async def wait(delay):
+    ...     await (time + delay)
+    ...
+    >>> run(wait(-20))  # delay until a time that has already passed
+
+When run regularly, this will fail with an :py:exc:`AssertionError`.
+The error includes some information to tell you what went wrong,
+and attempt to explain how to fix it.
+
+In addition to validating your simulation,
+assertion mode also provides rich exception messages.
+For example, erroneously using ``await time`` provides
+an Exception with additional help on intended usage:
+
+.. code:: none
+
+    TypeError: 'time' cannot be used in 'await' expression
+
+    Use 'time' to derive operands for specific expressions:
+    * 'await (time + duration)' to delay for a specific duration
+    * 'await (time == date)' to delay until a specific point in time
+    * 'await (time >= date)' to delay until after a point in time
```

Comment:
```suggestion
    * 'await (time >= date)' to delay until a point in time has passed
```

Reply:
This is a copy of the original error message. Should we change that one as well?

Reply:
You can also just ignore. It just reads a bit strange... You can decide.